simulator/fault_injection/CMakeLists.txt

```diff
@@ -21,6 +21,12 @@ ament_auto_add_library(fault_injection_node_component SHARED
   src/fault_injection_node/fault_injection_node.cpp
 )
 
+if(${rosidl_generator_cpp_VERSION} VERSION_LESS 2.5.0)
+  target_compile_definitions(fault_injection_node_component PRIVATE
+    USE_DEPRECATED_TO_YAML
```

Comment:
Rather than adding macros for each change, I prefer defining macros of ROS_DISTRO like autoware_auto_common.
https://gitlab.com/autowarefoundation/autoware.auto/AutowareAuto/blob/303ebe494891763d30eac1f2bb114396dc4d0404/src/tools/autoware_auto_cmake/cmake/autoware_auto_cmake.cmake#L42-47

I'll send a pull request to `autoware_common`.

@wep21 @esteve What do you think about it?

Reply:
Related: https://github.com/autowarefoundation/autoware.universe/pull/704/files#r856838564

cc @isamu-takagi

Reply:
@kenji-miyake I agree with you, but can I address your suggestion after merging this PR?

Reply:
I think it's okay. And I can work on the task.